Automattic/jetpack
# Blocks: Add AI-powered blocks for WordPress.com customers

#### Changes proposed in this Pull Request:

This PR introduces 2 new AI powered blocks for WordPress.com customers, per pbOQVh-2TF-p2 .

- The 2 blocks are now registered as Beta blocks, so they can be available to proxied a12s on WordPress.com simple sites and to folks using the Beta flag on WoA.
- The feature set is locked to WPCOM Simple / Atomic only, so no self-hosted sites.
- We now have 2 new wpcom v2 endpoints to proxy requests to WordPress.com to get data from CoAuthor.
- The backend work (lib) has already been deployed.

![Zrzut ekranu 2023-01-10 o 17 36 47](https://user-images.githubusercontent.com/3775068/211609443-53735b9a-194c-40ac-8a2f-f608db74844e.png)
![Zrzut ekranu 2023-01-10 o 17 36 37](https://user-images.githubusercontent.com/3775068/211609453-97547ae6-eada-4545-b7d7-11a31eb04b3e.png)
![Zrzut ekranu 2023-01-10 o 17 36 19](https://user-images.githubusercontent.com/3775068/211609454-89f615db-b751-4217-8fbe-3956c287d4e8.png)


**To Do**

The following can be done in follow-up PRs, but I think would be nice / necessary to implement.

- [ ] Write up a call4test / happiness heads-up posts
- [ ] Create a WordPress.com forums thread / form to gather feedback and link it in the post sidebar

#### Jetpack product discussion

Some Useful links: 

- Existing WordPress Coauthor Plugin: https://github.com/artpi/wp-writers-block-block 
- Primary issue in the Calypso repo: https://github.com/Automattic/wp-calypso/issues/71330
- PT: pbOQVh-2Ot-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:

1. Build and rsync to WPCOM or use `bin/jetpack-downloader test jetpack add/coauthor-block`
1. Apply following patch D97590-code
1. Go to Posts > Add New
1. **Test AI paragraph**
     1. Write a few paragraphs of your own text
     2. Search for "Jetpack AI Paragraph" block (or GPT), insert it 
     3. It should complete your text
1. **Test AI image**
     1. Insert "AI Image" image
     2. Write a prompt of what you would like to see
     3. Choose a variation
     4. block should be replaced by the generated image
1. On a Jurassic Ninja site, ensure the blocks are **NOT** available
